### PR TITLE
Add TSVREFE bit if temperature or internal reference channels is added

### PR DESCRIPTION
**Pull request by victorandrehc**
This PR enables TSVREFE bit automatically if either the internal temperature or internal voltage reference are added.

### CODE REVIEW

File: src/modm/platform/adc/stm32/adc_impl.hpp.in [hunk 1]
```diff
@@ -97,6 +97,12 @@ modm::platform::Adc{{ id }}::setChannel(const Channel channel,
 	ADC{{ per }}->SQR3 = uint32_t(channel) & 0x1f;
 
 	setSampleTime(channel, sampleTime);
+%% if temperature_available
+	if(channel == Channel::TemperatureSensor || channel==Channel::InternalReference)
+	{
+		ADC->CCR |= ADC_CCR_TSVREFE;
+	}
+%%endif
```
Comment (chris-durand) [inline src/modm/platform/adc/stm32/adc_impl.hpp.in]:
```suggestion
%% if temperature_available
	if(channel == Channel::TemperatureSensor || channel == Channel::InternalReference)
	{
		enableTemperatureRefVMeasurement();
	}
%% endif
```

You have probably missed [enableTemperatureRefVMeasurement()](https://github.com/modm-io/modm/blob/04ed0a52e03964811eb9764306bb1b61de5b5f60/src/modm/platform/adc/stm32/adc_impl.hpp.in#L47).

On F1 devices there is no `CCR` register. `enableTemperatureRefVMeasurement()` will always do the right thing.

File: src/modm/platform/adc/stm32/adc_impl.hpp.in [hunk 2]
```diff
@@ -129,6 +135,12 @@ modm::platform::Adc{{ id }}::addChannel(const Channel channel,
 	ADC{{ per }}->SQR1 = (ADC{{ per }}->SQR1 & ~ADC_SQR1_L) | (channel_count << 20);
 
 	setSampleTime(channel, sampleTime);
+%% if temperature_available
+	if(channel == Channel::TemperatureSensor || channel==Channel::InternalReference)
+	{
+		ADC->CCR |= ADC_CCR_TSVREFE;
+	}
+%%endif
```
Comment (chris-durand) [inline src/modm/platform/adc/stm32/adc_impl.hpp.in]:
```suggestion
%% if temperature_available
	if(channel == Channel::TemperatureSensor || channel == Channel::InternalReference)
	{
		enableTemperatureRefVMeasurement();
	}
%% endif
```

**Review by chris-durand — CHANGES_REQUESTED** [review]

File: src/modm/platform/adc/stm32/adc_impl.hpp.in (hunk 2 above)
Comment (victorandrehc) [reply src/modm/platform/adc/stm32/adc_impl.hpp.in]: just fixed it.

**Review by salkinium — APPROVED** [review]
Thanks!

**Review by rleh — APPROVED** [review]

**Review by chris-durand — APPROVED** [review]

**Comment by salkinium** [comment]
(rebased and will be merged next)

**Pull request merged**